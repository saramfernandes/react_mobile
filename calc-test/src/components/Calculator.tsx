import { add, sub, mul, div } from "../utils/math";
import React, { useState } from "react";
import { View, TextInput, Button, Text } from "react-native";

export default function Calculator() {
  const [a, setA] = useState("");
  const [b, setB] = useState("");
  const [result, setResult] = useState("");
  const toNum = (s: string) => Number(s.replace(",", "."));
  const calcula = (op: (x: number, y: number) => number) => {
    try {
      const na = toNum(a);
      const nb = toNum(b);
      const r = op(na, nb);
      setResult(String(r));
    } catch (e: any) {
      setResult(e?.message ?? "error");
    }
  };
  return (
    <View style={{ padding: 16 }}>
      <TextInput
        testID="input-a"
        placeholder="Número A"
        value={a}
        onChangeText={setA}
        keyboardType="numeric"
        style={{ borderWidth: 1, padding: 8, marginBottom: 8 }}
      />
      <TextInput
        testID="input-b"
        placeholder="Número B"
        value={b}
        onChangeText={setB}
        keyboardType="numeric"
        style={{ borderWidth: 1, padding: 8, marginBottom: 8 }}
      />
      <View
        style={{
          flexDirection: "row",
          justifyContent: "space-between",
          gap: 8,
        }}
      >
        <Button title="+" onPress={() => calcula(add)} />
        <Button title="-" onPress={() => calcula(sub)} />
        <Button title="×" onPress={() => calcula(mul)} />
        <Button title="÷" onPress={() => calcula(div)} />
      </View>
      <Text testID="result" style={{ marginTop: 16, fontSize: 18 }}>
        {result}
      </Text>
    </View>
  );
}
