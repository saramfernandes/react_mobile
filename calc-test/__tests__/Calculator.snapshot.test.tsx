import React from 'react';
import Calculator from '../src/components/Calculator';
import { render } from '@testing-library/react-native';

test('snapshot Calculator', () => {
 const tree = render(<Calculator />).toJSON();
 expect(tree).toMatchSnapshot();
});