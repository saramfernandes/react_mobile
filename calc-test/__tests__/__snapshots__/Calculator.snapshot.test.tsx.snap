// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Calculator 1`] = `
<View
  style={
    {
      "padding": 16,
    }
  }
>
  <TextInput
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Número A"
    style={
      {
        "borderWidth": 1,
        "marginBottom": 8,
        "padding": 8,
      }
    }
    testID="input-a"
    value=""
  />
  <TextInput
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Número B"
    style={
      {
        "borderWidth": 1,
        "marginBottom": 8,
        "padding": 8,
      }
    }
    testID="input-b"
    value=""
  />
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          +
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          -
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          ×
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          ÷
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "fontSize": 18,
        "marginTop": 16,
      }
    }
    testID="result"
  />
</View>
`;
